### PR TITLE
Fixes devices

### PR DESCRIPTION
**Pull request by fernandohds564**
Features:
 - add method to `PowerSupply` device to set strength and wait for monitor/readback to get there;
 - Remove auto-monitor of array PVs in `FamBPMs`;
 - Improve method `get_sampling_frequency` and make it a regular object method instead of a `staticmethod`
 - Improve logic of checking whether triggered acquisition finished sucessfully. Now the flow of a triggered acquisition should be like this:

```python3
fambpms = FamBPMs()
need_sum = True

ret = fambpms.mturn_config_acquisition()  #use as input here the necessary configurations
if ret < 0:
    raise Exception(f'BPM {-ret-1:d} did not finish last acquisition.')
elif ret >0:
    raise Exception(f'BPM {ret-1:d} is not ready for acquisition.')
fambpms.mturn_update_initial_orbit(consider_sum=need_sum)

# Send timing trigger

ret = fambpms.mturn_wait_update(consider_sum=need_sum)
if ret != 0:
    raise Exception(f'There was a problem with acquisition. Error code {ret:d}')

orbx, orby, sum = fambpms.get_mturn_orbit(return_sum=need_sum)

```

### CODE REVIEW

File: siriuspy/siriuspy/devices/pwrsupply.py [hunk 1]
```diff
@@ -203,6 +203,13 @@ def pwrstate(self, value):
         """."""
         self['PwrState-Sel'] = value
 
+    def set_strength(self, value, tol=0.2, timeout=10, wait_mon=False):
+        """Set RF phase and wait until it gets there."""
```
Comment (murilobalves) [inline siriuspy/siriuspy/devices/pwrsupply.py:207]: Copy-paste remnant in the docstring?

File: siriuspy/siriuspy/devices/pwrsupply.py [hunk 2]
```diff
@@ -203,6 +203,13 @@ def pwrstate(self, value):
...
+        self.strength = value
+        pv2wait = self._strength_rb_propty if wait_mon \
```
Comment (murilobalves) [inline siriuspy/siriuspy/devices/pwrsupply.py:209]: Shouldn't be `pv2wait = self._strength_mon_propty if wait_mon else self._strength_rb_propty`?

File: siriuspy/siriuspy/devices/bpm.py [hunk 3]
```diff
@@ -993,39 +998,86 @@ def mturn_config_acquisition(
...
+        Args:
+            wait (bool, optional): whether or not to wait BPMs get ready.
+                Defaults to True.
+            timeout (int, optional): Time to wait. Defaults to 10.
+
         Returns:
-            bool: Whether or not abort was successful.
+            int: code describing what happened:
+                =0: BPMs are ready.
+                >0: Index of the first BPM which did not updated plus 1.
```
Comment (murilobalves) [inline siriuspy/siriuspy/devices/bpm.py:1014]: updated -> update. This also appears in other docstrings

File: siriuspy/siriuspy/devices/pwrsupply.py (hunk 1 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/devices/pwrsupply.py]: Right. Done!

File: siriuspy/siriuspy/devices/pwrsupply.py (hunk 2 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/devices/pwrsupply.py]: Right. Done!

File: siriuspy/siriuspy/devices/bpm.py (hunk 3 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/devices/bpm.py]: Right. Done!

**Review by xresende — APPROVED** [review]

**Review by anacso17 — APPROVED** [review]

**Review by murilobalves — APPROVED** [review]

**Pull request merged**